keras/losses.py

```diff
@@ -92,16 +97,58 @@ def cosine_proximity(y_true, y_pred):
     y_pred = K.l2_normalize(y_pred, axis=-1)
     return -K.sum(y_true * y_pred, axis=-1)
 
+def root_mean_square_error(y_pred, targets):
```

Comment:
Not that square_root(a + b) != square_root(a) + square_root(b)

therefore square root cannot be used as a loss function, since we reduce per-batch losses via addition and subtraction.

This is also true for rms_log_error.